doc/guides/writing-tests.md

```diff
@@ -205,3 +205,15 @@ require('../common');
 const assert = require('assert');
 const freelist = require('internal/freelist');
 ```
+
+## Naming Test Files
+
+Test files are named using kebab casing. The first component of the name is
+`test`. The second is the module or subsystem being tested. The third is usually
+the method or event name being tested. Subsequent components of the name add
+more information about what is being tested.
+
+For example, a test for the `beforeExit` event on the `process` object might be
+named `test-process-beforeExit.js`. If the test specifically checked that arrow
```

Comment:
As per https://github.com/nodejs/node/pull/10581#issuecomment-270127932, we may have to discuss this further.